### PR TITLE
PRMP-935 new lambda-generate-stitch-record

### PR DESCRIPTION
**Pull request by NogaNHS**

### CODE REVIEW

File: infrastructure/lambda-generate-stitch-record.tf [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+      }
+    ]
+  })
+
+  depends_on = [module.generate-stitch-record-lambda, module.sns_encryption_key]
+}
+
+module "generate-stitch-record-lambda" {
+  source                   = "./modules/lambda"
+  name                     = "GenerateStitchRecord"
```
Comment (abbas-khan10) [inline infrastructure/lambda-generate-stitch-record.tf:45]: Can we call this `GenerateLloydGeorgeStitch` or `LloydGeorgeStitchJob`? So there is some consistency between this and the `LloydGeorgeStitch` lambda?

File: infrastructure/dynamo_db.tf [hunk 2]
```diff
@@ -134,6 +134,38 @@ module "zip_store_reference_dynamodb_table" {
   owner       = var.owner
 }
 
+module "stitch_store_reference_dynamodb_table" {
```
Comment (abbas-khan10) [inline infrastructure/dynamo_db.tf:137]: Can we change this name also, as there's isn't a `Stitch Store` bucket. Maybe 'LloydGeorgeStitchJobMetadata'?

File: infrastructure/lambda-delete-doc-ref.tf [hunk 3]
```diff
@@ -72,19 +72,21 @@ module "delete-doc-ref-lambda" {
     module.ndr-lloyd-george-store.s3_object_access_policy,
     "arn:aws:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
     "arn:aws:iam::aws:policy/CloudWatchLambdaInsightsExecutionRolePolicy",
-    module.ndr-app-config.app_config_policy_arn
+    module.ndr-app-config.app_config_policy_arn,
+    module.stitch_metadata_reference_dynamodb_table.dynamodb_policy
```
Comment (abbas-khan10) [inline infrastructure/lambda-delete-doc-ref.tf:76]: Are we missing a depends on for these?

File: infrastructure/dynamo_db.tf [hunk 4]
```diff
@@ -134,6 +134,38 @@ module "zip_store_reference_dynamodb_table" {
   owner       = var.owner
 }
 
+module "stitch_metadata_reference_dynamodb_table" {
+  source                      = "./modules/dynamo_db"
+  table_name                  = var.stitch_metadata_dynamodb_table_name
+  hash_key                    = "ID"
+  deletion_protection_enabled = false
```
Comment (chrisbloe) [inline infrastructure/dynamo_db.tf:141]: Probably should be `true`?

File: infrastructure/lambda-create-doc-ref.tf [hunk 5]
```diff
@@ -82,16 +83,17 @@ module "create-doc-ref-lambda" {
...
-    DOCUMENT_STORE_DYNAMODB_NAME = "${terraform.workspace}_${var.docstore_dynamodb_table_name}"
-    LLOYD_GEORGE_DYNAMODB_NAME   = "${terraform.workspace}_${var.lloyd_george_dynamodb_table_name}"
-    PDS_FHIR_IS_STUBBED          = local.is_sandbox,
-    WORKSPACE                    = terraform.workspace
-    PRESIGNED_ASSUME_ROLE        = aws_iam_role.create_post_presign_url_role.arn
+    STAGING_STORE_BUCKET_NAME     = "${terraform.workspace}-${var.staging_store_bucket_name}"
+    APPCONFIG_APPLICATION         = module.ndr-app-config.app_config_application_id
+    APPCONFIG_ENVIRONMENT         = module.ndr-app-config.app_config_environment_id
+    APPCONFIG_CONFIGURATION       = module.ndr-app-config.app_config_configuration_profile_id
+    DOCUMENT_STORE_BUCKET_NAME    = "${terraform.workspace}-${var.docstore_bucket_name}"
```
Comment (chrisbloe) [inline infrastructure/lambda-create-doc-ref.tf:90]: Hyphen?

File: infrastructure/lambda-create-doc-ref.tf [hunk 6]
```diff
@@ -101,6 +103,7 @@ module "create-doc-ref-lambda" {
     module.create-doc-ref-gateway,
     module.ndr-app-config,
     module.lloyd_george_reference_dynamodb_table,
-    module.document_reference_dynamodb_table
+    module.document_reference_dynamodb_table,
+    module.stitch_metadata_reference_dynamodb_table,
```
Comment (chrisbloe) [inline infrastructure/lambda-create-doc-ref.tf:107]: trailing comma

File: infrastructure/lambda-generate-document-manifest.tf [hunk 7]
```diff
@@ -55,7 +55,7 @@ module "generate-document-manifest-lambda" {
     "arn:aws:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
     "arn:aws:iam::aws:policy/CloudWatchLambdaInsightsExecutionRolePolicy",
     module.ndr-app-config.app_config_policy_arn,
-    aws_iam_policy.dynamodb_stream_policy.arn
+    aws_iam_policy.dynamodb_stream_manifest_policy.arn
```
[the remark this answers is not in the document]
Comment (chrisbloe) [reply infrastructure/lambda-generate-document-manifest.tf:58]: Why does it need to be in `depends_on`??

File: infrastructure/lambda-generate-document-manifest.tf [hunk 8]
```diff
@@ -78,11 +78,12 @@ module "generate-document-manifest-lambda" {
...
     module.ndr-zip-request-store,
     module.ndr-document-store,
-    module.ndr-lloyd-george-store
+    module.ndr-lloyd-george-store,
+    aws_iam_policy.dynamodb_stream_manifest_policy
   ]
 }
 
-resource "aws_iam_policy" "dynamodb_stream_policy" {
+resource "aws_iam_policy" "dynamodb_stream_manifest_policy" {
```
Comment (chrisbloe) [inline infrastructure/lambda-generate-document-manifest.tf:86]: If we're renaming this already, can it be called `dynamodb_stream_manifest` instead?

File: infrastructure/lambda-lloyd-george-record-stitch.tf [hunk 9]
```diff
@@ -67,37 +67,40 @@ module "lloyd-george-stitch-lambda" {
...
-    LLOYD_GEORGE_BUCKET_NAME   = "${terraform.workspace}-${var.lloyd_george_bucket_name}"
-    LLOYD_GEORGE_DYNAMODB_NAME = "${terraform.workspace}_${var.lloyd_george_dynamodb_table_name}"
-    CLOUDFRONT_URL             = module.cloudfront-distribution-lg.cloudfront_url
-    SPLUNK_SQS_QUEUE_URL       = try(module.sqs-splunk-queue[0].sqs_url, null)
-    WORKSPACE                  = terraform.workspace
-    PRESIGNED_ASSUME_ROLE      = aws_iam_role.stitch_presign_url_role.arn
+    APPCONFIG_APPLICATION         = module.ndr-app-config.app_config_application_id
+    APPCONFIG_ENVIRONMENT         = module.ndr-app-config.app_config_environment_id
+    APPCONFIG_CONFIGURATION       = module.ndr-app-config.app_config_configuration_profile_id
+    LLOYD_GEORGE_BUCKET_NAME      = "${terraform.workspace}-${var.lloyd_george_bucket_name}"
```
Comment (chrisbloe) [inline infrastructure/lambda-lloyd-george-record-stitch.tf:87]: Hyphen?

File: infrastructure/lambda-create-doc-ref.tf (hunk 5 above)
Comment (NogaNHS) [reply infrastructure/lambda-create-doc-ref.tf:90]: Yes, all of our s3 bucket names start with `${terraform.workspace}-`

**Review by chrisbloe — APPROVED** [review]
Thanks for making the changes :).

**Pull request merged**